jcl/src/java.base/share/classes/jdk/internal/misc/Unsafe.java

```diff
@@ -1573,10 +1573,15 @@ public int arrayBaseOffset(Class<?> c) {
 	 * 
 	 * @throws NullPointerException if class is null
 	 * @throws IllegalArgumentException if class is not an array
+	 * @throws RuntimeException if index scale is not a power of 2
 	 */
 	public int arrayIndexScale(Class<?> c) {
 		Objects.requireNonNull(c);
-		return arrayIndexScale0(c);
+		int indexScale = arrayIndexScale0(c);
+		if (indexScale <= 0 || (indexScale & (indexScale - 1)) != 0) {
```

Comment:
is `indexScale <= 0` possible?

We may be better off putting an assert in the create ramClass code

Reply:
It can be an assert that flattened instance size is less-or-equal than 0x7FFFFFFF

Reply:
Sounds good, is this the code you're talking about?
https://github.com/eclipse-openj9/openj9/blob/338146f5817f762f08e97f3a118075d5987e1017/runtime/vm/createramclass.cpp#L3339 
I could add an `Assert_VM_true(J9ARRAYCLASS_GET_STRIDE(ramClass) <= 0x7FFFFFFF)` right after the calls to `J9ARRAYCLASS_SET_STRIDE`

Reply:
yes, that will work

Reply:
addressed in a6d929c